packages/core/base/src/utils/ui.ts

```diff
@@ -7,41 +7,22 @@ export const getEnvironmentBaseUrl = (environment = ""): string => {
     return environment;
 };
 
-function blockchainTypeToLocatorStr(chain: string) {
-    switch (chain) {
-        case "solana":
-            return "sol";
-        case "ethereum":
-            return "eth";
-        case "polygon":
-            return "poly";
-        case "cardano":
-            return "ada";
-        case "bsc":
-            return "bsc";
-        default:
-            throw new Error(`Invalid chain type ${chain}`);
+function getNFTLocator(nft: NFT): string {
+    if (typeof nft === "string") {
+        return nft;
     }
-}
-
-function getNFTLocator(address: string, chain: string, tokenId?: string): string {
-    const chainLocator = blockchainTypeToLocatorStr(chain);
 
-    switch (chain) {
+    switch (nft.chain) {
         case "solana":
-            return `${chainLocator}:${address}`;
+            return `${ChainLocators.solana}:${nft.mintHash}`;
+        case "polygon":
         case "ethereum":
-            if (tokenId == null) {
-                throw new Error(
-                    `Missing or invalid tokenId when trying to get locator for NFT from contract ${address}`
-                );
-            }
-            return `${chainLocator}:${address}:${tokenId}`;
-        // address in this context is actually an AssetId
+        case "bsc":
+            return `${ChainLocators[nft.chain]}:${nft.contractAddress}:${nft.tokenId}`;
         case "cardano":
-            return `${chainLocator}:${address}`;
+            return `${ChainLocators[nft.chain]}:${nft.assetId}`;
         default:
-            throw new Error(`Invalid chain type ${chain}`);
+            throw new Error(`Invalid chain type ${JSON.stringify(nft)}`);
```

Comment:
if chain is wrong, we should only have nft.chain in the error, the other info is irrelevant for this